Features/outline

blforce opened the pull request:
Basic outline support based on @alexvdev's work.

alexvdev commented:
Great that you finalised my work and put the feature to trunk!
How can we test it?

blforce commented:
I updated the code that generates the guide so that it has outline bookmarks. I also added a short section on how to use the feature.

Is this an adequate method of testing? I didn't see any unit tests setup.

zoosky commented:
Ping @alafr

alafr commented:
Unfortunatelly I don't have enough time to install, test, and merge pull requests right now.
Did you try to make the changes requested here: https://github.com/devongovett/pdfkit/pull/216#issuecomment-39674038 ?

blforce commented:
@alafr I just modified my changes to follow the referenced comment. More closely. I didn't implement the `addSection` method and just return the PDFOutline object from the `addItem` method instead.

blforce commented:
Should be ready to go now. I've updated the documentation to reflect the new methods. Let me know if anything else needs updated.

alafr commented:
Thanks, it works great.
Just a few details:

- In my opinion the outline panel should be opened by default if the document has outlines. It can be done with `@_root.data.PageMode = 'UseOutlines'`.
- The `expanded` option is not described in the documentation.
- Is the `require('assert')` module in lib/mixins/outline.coffee still used?

blforce commented:
@alafr, I've made the requested changes.

blforce commented:
When is the next planned version release? I'd like to be able to use this with npm.

alafr commented:
I don't have access to npm, in the meantime you can replace with `npm install pdfkit` with `npm install https://github.com/devongovett/pdfkit/tarball/master`
